examples/webgpu_shadowmap.html

```diff
@@ -93,14 +93,35 @@
 
 				// geometry
 
-				const geometry = new THREE.TorusKnotGeometry( 25, 8, 75, 20 );
+				const geometry = new THREE.TorusKnotGeometry( 25, 8, 75, 80 );
 				const material = new THREE.MeshPhongMaterial( {
 					color: 0x999999,
 					shininess: 0,
 					specular: 0x222222
 				} );
 
-				torusKnot = new THREE.Mesh( geometry, material );
+				const materialCustomShadow = material.clone();
+			
+				const discardNode = vertexIndex.hash().greaterThan( 0.5 );
+
+				materialCustomShadow.colorNode = tslFn( () => {
+
+					discardNode.discard();
+
+					return vec4( 1. );
+
+				} )();
+
+				materialCustomShadow.shadowNode = tslFn( () => {
+
+					discardNode.discard();
+
+					return vec4( 1. );
+			
+				} )();
```

Comment:
I noticed that when commenting only this `shadowNode` line out neither the object surface nor the shadow have discard behavior even those the "colorNode" is still calling the discard node. Is this correct?

Reply:
Ah yes, it's because it wasn't a NodeMaterial (`MeshPhongNodeMaterial` instead of `MeshPhongMaterial`)